bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirCommandType.java

```diff
@@ -0,0 +1,862 @@
+        DecimalType decimalValue = value.as(DecimalType.class);
+
+        if (commandType != null && decimalValue != null) {
+            ComfoAirDataType dataType = commandType.getDataType();
+            if (dataType != null) {
+                int[] data = dataType.convertFromState(value, commandType);
+
+                if (data != null) {
+                    int dataPossition = commandType.getChangeDataPos();
+                    int intValue = decimalValue.intValue();
+
+                    return new ComfoAirCommand(key, commandType.change_command, null, data, dataPossition, intValue);
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get all commands which should be refreshed after a successful change
+     * command.
+     *
+     * @param key
+     *            command key
+     * @param usedKeys
+     * @return ComfoAirCommand's which should be updated after a modifying
+     *         ComfoAirCommand named by key
+     */
+    @SuppressWarnings("null")
+    public static Collection<ComfoAirCommand> getAffectedReadCommands(String key, Set<String> usedKeys) {
+
+        Map<Integer, ComfoAirCommand> commands = new HashMap<Integer, ComfoAirCommand>();
+
+        ComfoAirCommandType commandType = ComfoAirCommandType.getCommandTypeByKey(key);
+        if (commandType != null) {
+            if (commandType.read_reply_command != 0) {
+                Integer getCmd = commandType.read_command == 0 ? null : Integer.valueOf(commandType.read_command);
+                Integer replyCmd = Integer.valueOf(commandType.read_reply_command);
+
+                ComfoAirCommand command = new ComfoAirCommand(key, getCmd, replyCmd, new int[0], null, null);
+                commands.put(replyCmd, command);
+            }
+
+            if (commandType.change_affected != null) {
+                for (String affectedKey : commandType.change_affected) {
+                    // refresh affected event keys only when they are used
+                    if (!usedKeys.contains(affectedKey)) {
+                        continue;
+                    }
+
+                    ComfoAirCommandType affectedCommandType = ComfoAirCommandType.getCommandTypeByKey(affectedKey);
+
+                    if (affectedCommandType != null) {
+                        Integer getCmd = affectedCommandType.read_command == 0 ? null
+                                : Integer.valueOf(affectedCommandType.read_command);
+                        Integer replyCmd = Integer.valueOf(affectedCommandType.read_reply_command);
+
+                        ComfoAirCommand command = commands.get(replyCmd);
+
+                        if (command == null) {
+                            command = new ComfoAirCommand(affectedKey, getCmd, replyCmd, new int[0], null, null);
+                            commands.put(replyCmd, command);
+                        } else {
+                            command.addKey(affectedKey);
+                        }
```

Comment:
This logic is used in two places, so it would be better to refactor it into its own method.